distribution/config.go

```diff
@@ -142,9 +142,12 @@ func (s *imageConfigStore) RootFSFromConfig(c []byte) (*image.RootFS, error) {
 		return nil, err
 	}
 
-	// fail immediately on windows
+	// fail immediately on Windows when downloading a non-Windows image
+	// and vice versa
 	if runtime.GOOS == "windows" && unmarshalledConfig.OS == "linux" {
 		return nil, fmt.Errorf("image operating system %q cannot be used on this platform", unmarshalledConfig.OS)
+	} else if runtime.GOOS != "windows" && unmarshalledConfig.OS == "windows" {
```

Comment:
> why not just check `runtime.GOOS != unmarshelledConfig.OS` here?

I'm afraid this might have some unintended consequences. For example, what if there are images out there built with nonstandard tools that don't set the `OS` field? For now I think it's best to explicitly check for incompatible configurations.